packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -566,16 +580,17 @@ export function getWorkflowHooksMain(
 	data: IWorkflowExecutionDataProcess,
 	executionId: string,
 ): WorkflowHooks {
+	const { pushRef, retryOf } = data;
+	const saveSettings = toSaveSettings(data.workflowData.settings);
+	const optionalParameters = { pushRef, retryOf: retryOf ?? undefined, saveSettings };
```

Comment:
```suggestion
	const optionalParameters = { pushRef, retryOf, saveSettings };
```

Reply:
still not confident enough in the typings here